### PR TITLE
Detach loss tensor before scalar conversion

### PR DESCRIPTION
**Pull request by drivanov**
This PR addresses a PyTorch UserWarning in TGN examples, such as the one shown below:
```
UserWarning: Converting a tensor with requires_grad=True to a scalar may lead to unexpected behavior.
Consider using tensor.detach() first.
(Triggered internally at /opt/pytorch/pytorch/torch/csrc/autograd/generated/python_variable_methods.cpp:835.)
total_loss += float(loss)
```
**Issue:**
The warning is triggered when attempting to convert a loss tensor (with requires_grad=True) directly into a Python float using `float(loss)`. This may inadvertently break the computation graph or lead to subtle bugs in gradient tracking.

**Fix:**
The loss tensor is now explicitly detached from the computation graph before converting to a scalar using:
```
total_loss += float(loss.detach())
```

**Impact:**
This suppresses the warning and ensures the scalar conversion is safe and free from unintended autograd behavior. The fix does not alter model training behavior or loss values.

### CODE REVIEW

**Comment by puririshi98** [comment]
@shenyangHuang let us know if anything else needed to merge

**Review by shenyangHuang — APPROVED** [review]
Looks good
